### PR TITLE
Add a check for bad filenames in the directory containing the user's code

### PR DESCRIPTION
**Pull request by schmave**
This is a draft of a solution that could work. What do you think? Here's a sample error message:

```
(venv) Evans-MacBook-Pro:test-cmu evan$ python test.py 

******************************************************************************
* Warning: The following files in /Users/evan/test-cmu
* may prevent your program from running correctly. Please rename these files.
*
* select.py
******************************************************************************

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 69, in <module>
    import msvcrt
ModuleNotFoundError: No module named 'msvcrt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/evan/test-cmu/test.py", line 1, in <module>
    from cmu_graphics import *
  File "/Users/evan/test-cmu/venv/lib/python3.10/site-packages/cmu_graphics/__init__.py", line 160, in <module>
    from .libs import loader_util
  File "/Users/evan/test-cmu/venv/lib/python3.10/site-packages/cmu_graphics/libs/loader_util.py", line 2, in <module>
    import platform
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/platform.py", line 119, in <module>
    import subprocess
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 75, in <module>
    import select
  File "/Users/evan/test-cmu/select.py", line 6, in <module>
    cmu_graphics.run()
NameError: name 'cmu_graphics' is not defined
```

### CODE REVIEW

File: cmu_graphics/__init__.py [hunk 1]
```diff
@@ -1,4 +1,56 @@
+import __main__
+import os
+
+main_directory = os.path.dirname(__main__.__file__)
+main_sibling_filenames = set(os.listdir())
```
Comment (austin-schick) [inline cmu_graphics/__init__.py:5]: I think we probably want to use `main_directory` in this call to `os.listdir()`. The docs say: "The first entry in the module search path is the directory that contains the input script". So we do want `main_directory`, not the current working directory.

**Comment by austin-schick** [comment]
Looks pretty good to me! What do you think about exiting immediately when we print the warning? Some arguments for that behavior:

* For some of these modules, we're pretty confident that the bad filename is going to cause a crash anyway
* Our warning could become lost above the stack trace when the program exits
* It might be nicer to fail immediately instead of waiting and failing in a confusing way down the line

I don't have a super strong opinion either way, but I do think it's worth thinking about.

**Comment by dkosbie** [comment]
How about both, main_directory and cwd?  Also, I think we should hide the
rest of the error message (the stack trace), so they only see the part
telling them to rename or remove the offending file (and I think we should
say that, "rename or remove").  Thx.

**Comment by dkosbie** [comment]
Right, what Austin just said.  Clear early fail.

**Comment by austin-schick** [comment]
@palascat100 and @josht04 I think this is ready for you to test. I'm leaving out Alex because he's on Linux, which Desktop CMU Graphics doesn't support.

**Review by palascat100 — APPROVED** [review]
Looks good to me!

**Review by josht04 — APPROVED** [review]
Also works for me :)

**Comment by austin-schick** [comment]
Thanks!

**Pull request merged**